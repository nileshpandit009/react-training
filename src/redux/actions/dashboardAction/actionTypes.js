export const DASHBOARD_CONSTANTS = {
  FETCH_POSTS: "FETCH_POSTS",
  FETCH_POSTS_SUCCESS: "FETCH_POSTS_SUCCESS",
  FETCH_POSTS_ERROR: "FETCH_POSTS_ERROR",

  FETCH_POST_DETAILS: "FETCH_POST_DETAILS",
  FETCH_POST_DETAILS_SUCCESS: "FETCH_POST_DETAILS_SUCCESS",
  FETCH_POST_DETAILS_ERROR: "FETCH_POST_DETAILS_ERROR",

  RESET_POSTS: "RESET_POSTS",
  RESET_POST_DETAILS: "RESET_POST_DETAILS",
};
