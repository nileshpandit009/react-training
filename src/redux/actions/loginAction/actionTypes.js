export const LOGIN_CONSTANTS = {
  EMAIL: "EMAIL",
  PASSWORD: "PASSWORD",
  EMAIL_ERR: "EMAIL_ERR",
  PASSWORD_ERR: "PASSWORD_ERR",
  LOGIN_ERR: "LOGIN_ERR",
  RESET_LOGIN: "RESET_LOGIN",
  LOGIN_REQUEST: "LOGIN_REQUEST",
  LOGIN_SUCCESS: "LOGIN_SUCCESS",
  LOGIN_FAILURE: "LOGIN_FAILURE",
};
