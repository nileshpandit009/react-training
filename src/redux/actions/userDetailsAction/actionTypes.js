export const USER_DETAILS_CONSTANTS = {
  USER_DETAILS_REQUEST: "USER_DETAILS_REQUEST",
  USER_DETAILS_SUCCESS: "USER_DETAILS_SUCCESS",
  USER_DETAILS_FAILURE: "USER_DETAILS_FAILURE",
  RESET_USER_DETAILS: "RESET_USER_DETAILS",
};
