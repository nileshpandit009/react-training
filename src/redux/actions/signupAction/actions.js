import { SIGNUP_CONSTANTS } from "./actionTypes";

const { SIGNUP_REQUEST, SIGNUP_SUCCESS, SIGNUP_FAILURE } = SIGNUP_CONSTANTS;

export const signupRequest = (payload) => ({
  type: SIGNUP_REQUEST,
  payload,
});

export const signupSuccess = (payload) => ({
  type: SIGNUP_SUCCESS,
  payload,
});

export const signupFailure = (payload) => ({
  type: SIGNUP_FAILURE,
  payload,
});
