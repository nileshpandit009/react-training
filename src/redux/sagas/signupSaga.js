import { call, put, takeEvery } from "@redux-saga/core/effects";
import { doSignup } from "apis/signupApi";
import {
  signupFailure,
  signupSuccess,
} from "redux/actions/signupAction/actions";
import { SIGNUP_CONSTANTS } from "redux/actions/signupAction/actionTypes";

function* makeSignupRequest(action) {
  const { formData } = action.payload;
  try {
    const response = yield call(doSignup, formData);
    yield put(
      signupSuccess({
        success: "User registration successful!",
        uuid: response.data.uuid,
      })
    );
  } catch (e) {
    const { response } = e;
    yield put(signupFailure({ error: response.data._error_message }));
  }
}

export function* signupSaga() {
  yield takeEvery(SIGNUP_CONSTANTS.SIGNUP_REQUEST, makeSignupRequest);
}
