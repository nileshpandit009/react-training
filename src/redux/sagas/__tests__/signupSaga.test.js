import { put, takeEvery } from "@redux-saga/core/effects";
import {
  signupFailure,
  signupRequest,
  signupSuccess,
} from "redux/actions/signupAction/actions";
import { SIGNUP_CONSTANTS } from "redux/actions/signupAction/actionTypes";
import { makeSignupRequest, signupSaga } from "../signupSaga";

describe("Test signup saga", () => {
  let makeSignupRequestGenerator;
  beforeEach(() => {
    const payload = {
      formData: {},
    };
    makeSignupRequestGenerator = makeSignupRequest(signupRequest(payload));
    const callDescriptor = makeSignupRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it("must call signupSuccess action", () => {
    const response = {
      data: {
        success: "User registration successful!",
        uuid: "some uuid for user object",
      },
    };

    const putDescriptor = makeSignupRequestGenerator.next(response).value;

    expect(putDescriptor).toEqual(
      put(
        signupSuccess({
          success: "User registration successful!",
          uuid: response.data.uuid,
        })
      )
    );

    expect(makeSignupRequestGenerator.next().done).toBeTruthy();
  });

  it("must call signupFailure action", () => {
    const errResponse = {
      status: 400,
      response: {
        data: {
          _error_message: "some error message",
        },
      },
    };

    const putDescriptor = makeSignupRequestGenerator.throw(errResponse).value;
    expect(putDescriptor).toEqual(
      put(signupFailure({ error: errResponse.response.data._error_message }))
    );

    expect(makeSignupRequestGenerator.next().done).toBeTruthy();
  });
});

describe("Test signup saga watchers", () => {
  let signupSagaItr;
  beforeAll(() => {
    signupSagaItr = signupSaga();
  });

  it("must start task to watch for makeSignupRequest action", () => {
    const takeEveryDescriptor = signupSagaItr.next().value;
    expect(takeEveryDescriptor).toEqual(
      takeEvery(SIGNUP_CONSTANTS.SIGNUP_REQUEST, makeSignupRequest)
    );

    expect(signupSagaItr.next().done).toBeTruthy();
  });
});
