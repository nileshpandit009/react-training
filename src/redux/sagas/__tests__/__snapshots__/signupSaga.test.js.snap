// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test signup saga must call signupFailure action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`Test signup saga must call signupSuccess action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
